en/components/for_of.md

```diff
@@ -0,0 +1,156 @@
+            let-rowIndex="index">
+                <td  [style.width.px]='item.width' style='height:100px;'>{{rowIndex}} : {{item.text}}</td>
+            </ng-template>
+    </tr>
+    </tbody>
+</table>
+```
+
+#### Horizontal and vertical virtualization
+
+```html
+<div #container [style.width]='width' [style.height]='height'>
+    <ng-template #scrollContainer igxFor let-rowData [igxForOf]="data"
+        [igxForScrollOrientation]="'vertical'"
+        [igxForContainerSize]='height'
+        [igxForItemSize]='"50px"'>
+        <div [style.display]="'flex'" [style.height]="'50px'">
+            <ng-template #childContainer igxFor let-col [igxForOf]="cols"
+                [igxForScrollOrientation]="'horizontal'"
+                [igxForScrollContainer]="parentVirtDir"
+                [igxForContainerSize]='width'>
+                    <div [style.min-width]='col.width + "px"'>{{rowData[col.field]}}</div>
+            </ng-template>
+        </div>
+    </ng-template>
+</div>
+```
+
+**Note**: The `igxForOf` directive can bind to remote service too. Review the following demo for more information.
+#### igxFor bound to remote service Demo
+<div class="sample-container loading" style="height:780px">
+    <iframe src='{environment:demosBaseUrl}/igx-for-sample-2' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<br/>
+<div class="divider--half"></div>
+
+## API
+
+### Inputs
+
+Below is the list of all inputs that the developers may set to configure the `igxFor` look/behavior:
+| Name | Type | Description |
+| :--- |:--- | :--- |
+| id | string | Unique identifier of the Grid |
+| `igxForOf` | any[] | The data to be virtualized |
+| `igxForScrollOrientation` | string | Virtualization direction - "horizontal" or "vertical" |
+| `igxForScrollContainer` | any | The container where the vertical and horizontal scrollbars will be created, the is useful when nesting the directive and for cases where the scrolling container is not going to be the direct parent |
+| `igxForContainerSize` | any | Specifies the container size |
+| `igxForItemSize` | any | Specifies the item size, when the virtualization is vertical it is used as height and as width when the virtualization is horizontal. It is mostly used for the vertical direction, because for the horizontal width it is possible to have items with different widhts |
+
+<div class="divider--half"></div>
+
+### Outputs
+
+A list of the events emitted by the **igx-for**:
+
+| Name              | Description                                                          |
+| :---------------- | :------------------------------------------------------------------- |
+| _Event emitters_  | _Notify for a change_                                                |
+| `onItemLoad`      | Used when scrolled to emit the loaded data item                      |
+| `onItemPreload`   | Used when scrolled to emit the data item which is about to be loaded |
```

Comment:
I'm waiting on agreement on the PR to the main repository to approve the final API.
https://github.com/IgniteUI/igniteui-angular/pull/619